frontend/src/metabase-lib/metadata.ts

```diff
@@ -143,6 +144,11 @@ declare function DisplayInfoFn(
   stageIndex: number,
   segment: SegmentMetadata,
 ): SegmentDisplayInfo;
+declare function DisplayInfoFn(
+  query: Query,
+  stageIndex: number,
+  queryInfo: Query,
```

Comment:
I agree with @kamilmielnik here, but I'm not sure oveloading will work this way - it's only type definition, the function itself needs to be adopted as well and we need to pass some dumb values as first two params anyway,

so probalby keeping 3 arguments anyway is kind of a workaround, but it deserves a comment

Reply:
> I don't understand why this function gets 3 arguments

It accepts three.
- _query
- _stage-number
- query

@bshepherdson explains it here:
https://metaboat.slack.com/archives/C0645JP1W81/p1705424317194709?thread_ts=1705406252.958099&cid=C0645JP1W81

> you can call `display-info` with all kinds of things as the third argument, and some of them need the query + stage context to look up metadata and get things like display names.

Reply:
as I understand Kamil's point, we can pass anything to the ML function, so it can be `display_info(null, null, query)` and it means we can simplify it on FE to `displayInfo(query)` which will be a wrapper for `display_info(null, null, query), but it's going to be another function and I don't see a clean way right away how to achieve it without an additional wrapper or if statement

Reply:
I'm sorry to break the convo here, but I had to split the PR so the relevant bit now lives in:
https://github.com/metabase/metabase/pull/37786